src/containers/List/ListWrapper.tsx

```diff
@@ -84,6 +85,24 @@ const ListWrapper: React.FC = () => {
     }
   }
 
+  const handleExpansion = (serialNumber: string) => {
+    if (!applicationsRows) return
+    const findApplication = applicationsRows.find(
+      ({ application }) => application.serial === serialNumber
+    )
+    if (findApplication) {
+      const { expanded } = findApplication
+      findApplication.expanded = !expanded
+    }
+    setApplicationsRows(applicationsRows)
+    setReloadTable(true)
```

Comment:
Ah, I see -- you've used the "ReloadTable" value to handle the re-rendering problem you were showing me earlier.

However, I think the source of the problem is that `setApplicationsRows(applicationsRows)` is just setting `applicationRows` with itself. And it hasn't changed -- the values inside have but it's just pointing to the same object reference.

Anyway, I just tested it and it will work if you just spread the array to force a new copy:
`setApplicationsRows([...applicationsRows])`

Then you can do away with the whole reloadTable value completely.

Also -- why not pass the whole Application (rather than just the serial) to the `handleExpansion` method, then you won't have to do the `.find`.

Then the whole method could be reduced to (this also assumes you have the "isExpanded" field on the main application object, as suggested above):
```
  const handleExpansion = (application: ApplicationListRow) => {
    if (!applicationsRows) return
    application.isExpanded = !(application.isExpanded) // Updates in place inside applicationRows
    setApplicationsRows([...applicationsRows])
  }
```

Reply:
I get a bit confused with this difference on updating the object properties which doesn't change the object itself. And didn't know about this option to do the spread to force the  a new copy. Thank you so much for this suggestion. I'll implement it now